### ISSUE TITLE
New supervisor fails to load in networks without Internet access 

### ISSUE THREAD

**Issue by ericcalabretta** [habitat-sh/habitat#7010]
Newer version of the supervisor are unable to start if they can't reach googles DNS and if `HAB_OUTBOUND_IP_ADDR_LOOKUP_SOCKET_ADDR` is not set. Habitat will fail to launch with `Launcher exiting with code 86` 

Previous version of the supervisor would warn `hab-sup(SY): IP Address lookup failed; using fallback of 127.0.0.1 (A socket operation was attempted to an unreachable network. (os error 10051))` but would continue to start. 

- [ ] The OS (including version) where you are running any of the Habitat commands.
Windows 7 and 10

- [ ] Debug/backtrace of the command you are trying to run. You can set the following environment variables before running the `hab` command to generate a trace:

```
2019-10-01 14:04:25,301 - Habitat windows service is starting launcher at: C:\hab\pkgs\core\hab-launcher\11645\20190712230228\bin\hab-launch.exe
2019-10-01 14:04:25,320 - Habitat windows service is starting launcher with args: run --no-color --peer IP --url URL --health-check-interval 300 --event-stream-application=app --event-stream-environment=env --event-stream-site=site --event-stream-url=url --event-stream-token=token --event-stream-connect-timeout=0
2019-10-01 14:04:25,566 - hab-sup(MR): core/hab-sup (core/hab-sup/0.87.0-dev/20190930145300)
2019-10-01 14:04:25,570 - Failed to discover this host's outbound IP address: A socket operation was attempted to an unreachable network. (os error 10051)
2019-10-01 14:04:25,596 - Logging configuration file 'C:\hab/sup\default\config\log.yml' not found; using default logging configuration
2019-10-01 14:04:25,596 -   0 Enabling feature: IGNORE_SIGNALS
2019-10-01 14:04:25,596 -   0 Enabling feature: EVENT_STREAM
2019-10-01 14:04:25,681 - [2019-10-01T18:04:25Z ERROR hab_launch] Launcher exiting with code 86
```
		
This may be related to: 
https://github.com/habitat-sh/habitat/pull/6953

**Comment by gscho** [comment]
Not sure if this is easily done or not but it might be nice to try connecting to <peer>:9638 if the peer flag is supplied. My understanding is that it can us any UDP connection to determine the outbound IP.

**Comment by davidMcneil** [comment]
Closed by #6960
